src/components/include/DefaultImportModalDialog.js

```diff
@@ -0,0 +1,177 @@
+        b: 500.0,
+        c: 500.0,
+        alpha: 90,
+        beta: 90,
+        gamma: 90,
+        units: {
+            length: Made.units.angstrom,
+            angle: Made.units.degree
+        }
+    }
+};
+
+const mockLatticeConstantMultiplicator = 2.0;
+
+class DefaultImportModalDialog extends ModalDialog {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            xyz: "Paste your XYZ"
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleFileRead = this.handleFileRead.bind(this);
+
+        this.reader = new FileReader();
+        this.reader.onloadend = this.handleFileRead;
+    }
+
+    handleFileRead(evt) {
+        this.setState({xyz: evt.target.result});
+    }
+
+    handleSubmit() {
+
+        if (!this.BasisTextComponent.state.isContentValidated) return;
+
+        const basisConfig = Made.parsers.xyz.toBasisConfig(this.state.xyz, Made.ATOMIC_COORD_UNITS.cartesian);
+
+        // determine the max distance to use for cell for user's XYZ
+        const x = [],
+            y = [],
+            z = [];
+        basisConfig.coordinates.forEach(function(item){
+            x.push(item.value[0]);
+            y.push(item.value[1]);
+            z.push(item.value[2]);
+        });
+
+        const diffX = _.max(x) - _.min(x),
+            diffY = _.max(y) - _.min(y),
+            diffZ = _.max(z) - _.min(z);
+        const mockLatticeConstant = _.max([diffX, diffY, diffZ]) * mockLatticeConstantMultiplicator;
+
+        const material = new Made.Material(defaultXYZMaterialConfig);
+
+        const latticeConfig = Object.assign({}, material.lattice, {
+            a: mockLatticeConstant,
+            b: mockLatticeConstant,
+            c: mockLatticeConstant
+        });
+        const lattice = new Made.Lattice(latticeConfig);
+
+        const basis = new Made.Basis({
+            ...basisConfig,
+            cell: lattice.vectorArrays
+        });
+
+        const newMaterialConfig = Object.assign({},
+            material.toJSON(),
+            {
+                basis: basis.toJSON(),
+                lattice: lattice.toJSON(),
+                name: `${material.name} - ${basis.formula}`
+            }
+        );
+        const newMaterial = new Material(newMaterialConfig);
+        newMaterial.cleanOnCopy();
+
+        this.props.onSubmit([newMaterial]);
+    }
+
+    handleChange(content) {
+        this.setState({xyz: content});
+    }
+
+    handleFileChange(files) {
+        if (!files[0] || !files[0].size) return alert("Error: file cannot be read (unaccessible?)");
+        this.reader.currentFilename = files[0].name;
+        this.reader.readAsText(files[0]);
+    }
+
+    renderHeader() {
+        return (
+            <ModalHeader className="bgm-dark" closeButton={true}>
+                <h4 className="modal-title">{this.props.title}
```

Comment:
We should have a proper title and remove the link on the lines below - the combinatorial basis concept needs more explanation, here we don't need it